src/main/scala/mesosphere/marathon/storage/repository/legacy/store/MesosStateStore.scala

```diff
@@ -34,17 +39,23 @@ class MesosStateStore(state: State, timeout: Duration) extends PersistentStore {
       .recover(mapException(s"Can not create entity with key $key"))
   }
 
-  override def update(entity: PersistentEntity): Future[PersistentEntity] = entity match {
-    case MesosStateEntity(id, v) =>
-      futureToFuture(state.store(v))
-        .recover(mapException(s"Can not update entity with key ${entity.id}"))
-        .map(throwOnNull)
-        .map(MesosStateEntity(id, _))
+  override def update(entity: PersistentEntity): Future[PersistentEntity] = {
+    require(isOpen, "the store must be opened before it can be used")
+
+    entity match {
+      case MesosStateEntity(id, v) =>
+        futureToFuture(state.store(v))
+          .recover(mapException(s"Can not update entity with key ${entity.id}"))
+          .map(throwOnNull)
+          .map(MesosStateEntity(id, _))
 
-    case _ => throw new IllegalArgumentException("Can not handle this kind of entity")
+      case _ => throw new IllegalArgumentException("Can not handle this kind of entity")
```

Comment:
Can we use `Future.failed(new IllegalArgumentException("Can not handle this kind of entity"))` here?

Reply:
I am hesitant to change it because I might be unaware what intention the original author had. I my guess is it should not normally happen, but if it happens, just fail loudly pointing out that the code is incorrect in some way.